### PR TITLE
fix: Forbid access to files outside source control work directory

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Forbid access to files outside source control work directory

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/SEC-144/path-traversal-cwe-23

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/environments/source-control/source-control-helper.ee.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import { generateKeyPairSync } from 'crypto';
 import { constants as fsConstants, mkdirSync, accessSync } from 'fs';
+import { ApplicationError } from 'n8n-workflow';
+import { ok } from 'node:assert/strict';
```
Comment (valya) [inline packages/cli/src/environments/source-control/source-control-helper.ee.ts:4]: Not something for this PR, but do we maybe want to investigate a more flexible solution for asserts? Or at least set some guidelines (e.g. whether to use `assert/strict` or not).

**Review by valya — APPROVED** [review]

File: packages/cli/src/utils/path-util.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as path from 'node:path';
+
+/**
+ * Checks if the given childPath is contained within the parentPath. Resolves
+ * the paths before comparing them, so that relative paths are also supported.
+ */
+export function isContainedWithin(parentPath: string, childPath: string): boolean {
```
Comment (valya) [inline packages/cli/src/utils/path-util.ts:7]: We should probably look into other places that need this at some point

Comment (tomi) [reply packages/cli/src/utils/path-util.ts:7]: Definitely. There's at least on in `Server.ts` that I'll update after this is merged

File: packages/cli/src/environments/source-control/source-control-helper.ee.ts (hunk 1 above)
Comment (tomi) [reply packages/cli/src/environments/source-control/source-control-helper.ee.ts:4]: Good thinking. I think for 99% of the cases we are checking
1. if something is defined and not `null` or `undefined` or
2. if some condition is true.

I think for that `node:assert` is fulfils the need. But yeah would be good to align.

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Comment by janober** [comment]
Got released with [`n8n@1.63.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.63.0)

**Pull request merged**